File: frontend/src/services/msuiteAccountService.ts
import api from './api'
import { MsuiteAccount, MsuiteAccountRequest, MsuiteCountry } from '../types'

export const msuiteAccountService = {
  async getAll(): Promise<MsuiteAccount[]> {
    const response = await api.get<MsuiteAccount[]>('/msuite-accounts')
    return response.data
  },

  async getByCountry(country: MsuiteCountry): Promise<MsuiteAccount[]> {
    const response = await api.get<MsuiteAccount[]>(`/msuite-accounts/by-country/${country}`)
    return response.data
  },

  async getById(id: number): Promise<MsuiteAccount> {
    const response = await api.get<MsuiteAccount>(`/msuite-accounts/${id}`)
    return response.data
  },

  async create(request: MsuiteAccountRequest): Promise<MsuiteAccount> {
    const response = await api.post<MsuiteAccount>('/msuite-accounts', request)
    return response.data
  },

  async update(id: number, request: MsuiteAccountRequest): Promise<MsuiteAccount> {
    const response = await api.put<MsuiteAccount>(`/msuite-accounts/${id}`, request)
    return response.data
  },

  async delete(id: number): Promise<void> {
    await api.delete(`/msuite-accounts/${id}`)
  },
}

File: backend/src/main/java/com/personalweb/app/service/MsuiteAccountService.java
package com.personalweb.app.service;

import com.personalweb.app.dto.MsuiteAccountDto;
import com.personalweb.app.dto.MsuiteAccountRequest;
import com.personalweb.app.entity.MsuiteAccount;

import java.util.List;

public interface MsuiteAccountService {

    List<MsuiteAccountDto> getAll();

    List<MsuiteAccountDto> getByCountry(MsuiteAccount.Country country);

    MsuiteAccountDto getById(Long id);

    MsuiteAccountDto create(MsuiteAccountRequest request);

    MsuiteAccountDto update(Long id, MsuiteAccountRequest request);

    void delete(Long id);
}
